Test failure: `TestGetSetSystemAgentManagerConfiguration`

```
GOROOT=/usr/local/go #gosetup
GOPATH=/Users/cmarget/go #gosetup
/usr/local/go/bin/go test -c --tags=integration -tags integration -o /Users/cmarget/Library/Caches/JetBrains/GoLand2024.3/tmp/GoLand/___TestGetSetSystemAgentManagerConfiguration_in_github_com_Juniper_apstra_go_sdk_apstra.test github.com/Juniper/apstra-go-sdk/apstra #gosetup
/usr/local/go/bin/go tool test2json -t /Users/cmarget/Library/Caches/JetBrains/GoLand2024.3/tmp/GoLand/___TestGetSetSystemAgentManagerConfiguration_in_github_com_Juniper_apstra_go_sdk_apstra.test -test.v=test2json -test.paniconexit0 -test.run ^\QTestGetSetSystemAgentManagerConfiguration\E$ #gosetup
=== RUN   TestGetSetSystemAgentManagerConfiguration
2024/12/26 15:22:26 unsupported API version: '5.1.0'
=== RUN   TestGetSetSystemAgentManagerConfiguration/aws/i-0a36daf0e2501f5e7/5.1.0
=== PAUSE TestGetSetSystemAgentManagerConfiguration/aws/i-0a36daf0e2501f5e7/5.1.0
=== CONT  TestGetSetSystemAgentManagerConfiguration/aws/i-0a36daf0e2501f5e7/5.1.0
2024/12/26 15:22:26 testing GetSystemAgentManagerConfig() against aws i-0a36daf0e2501f5e7 (5.1.0)
2024/12/26 15:22:26 testing SetSystemAgentManagerConfig() against aws i-0a36daf0e2501f5e7 (5.1.0)
    api_system_agent_test.go:49: 
        	Error Trace:	/Users/cmarget/projects/work/apstra-go-sdk/apstra/api_system_agent_test.go:49
        	Error:      	Received unexpected error:
        	            	{"errors": {"device_os_image_download_timeout": "Value is required"}} - http response '422 UNPROCESSABLE ENTITY' at 'https://i-5-1-0-117.t3aco.fragmentationneeded.net/api/system-agent/manager-config?async=full'
        	Test:       	TestGetSetSystemAgentManagerConfiguration/aws/i-0a36daf0e2501f5e7/5.1.0
--- FAIL: TestGetSetSystemAgentManagerConfiguration/aws/i-0a36daf0e2501f5e7/5.1.0 (0.29s)

--- FAIL: TestGetSetSystemAgentManagerConfiguration (0.59s)

FAIL

Process finished with the exit code 1
```

--- a/apstra/api_system_agent.go
+++ b/apstra/api_system_agent.go
@@ -12,6 +12,7 @@ import (
 const apiUrlSystemAgentManagerConfig = "/api/system-agent/manager-config"
 
 type SystemAgentManagerConfig struct {
+	DeviceOsImageDownloadTimeout    *int `json:"device_os_image_download_timeout"` // introduced in and required by 5.1.0 (1-2700)
 	SkipRevertToPristineOnUninstall bool `json:"skip_revert_to_pristine_on_uninstall"`
 	SkipPristineValidation          bool `json:"skip_pristine_validation"`
 	SkipInterfaceShutdownOnUpgrade  bool `json:"skip_interface_shutdown_on_upgrade"`

--- a/apstra/api_system_agent.go
+++ b/apstra/api_system_agent.go
@@ -12,7 +12,7 @@ import (
 const apiUrlSystemAgentManagerConfig = "/api/system-agent/manager-config"
 
 type SystemAgentManagerConfig struct {
-	DeviceOsImageDownloadTimeout    *int `json:"device_os_image_download_timeout"` // introduced in and required by 5.1.0 (1-2700)
+	DeviceOsImageDownloadTimeout    *int `json:"device_os_image_download_timeout,omitempty"` // introduced in and required by 5.1.0 (1-2700)
 	SkipRevertToPristineOnUninstall bool `json:"skip_revert_to_pristine_on_uninstall"`
 	SkipPristineValidation          bool `json:"skip_pristine_validation"`
 	SkipInterfaceShutdownOnUpgrade  bool `json:"skip_interface_shutdown_on_upgrade"`

--- a/apstra/api_system_agent_test.go
+++ b/apstra/api_system_agent_test.go
@@ -10,6 +10,7 @@ package apstra
 import (
 	"context"
 	"log"
+	"math/rand/v2"
 	"testing"
 
 	"github.com/Juniper/apstra-go-sdk/apstra/compatibility"
@@ -34,6 +35,9 @@ func TestGetSetSystemAgentManagerConfiguration(t *testing.T) {
 				SkipRevertToPristineOnUninstall: !mgrCfg.SkipRevertToPristineOnUninstall,
 				SkipPristineValidation:          !mgrCfg.SkipPristineValidation,
 				SkipInterfaceShutdownOnUpgrade:  !mgrCfg.SkipInterfaceShutdownOnUpgrade,
+			}
+			if compatibility.HasDeviceOsImageDownloadTimeout.Check(client.client.apiVersion) {
+				testCfg.DeviceOsImageDownloadTimeout = toPtr(rand.IntN(2700) + 1)
 			}
 
 			// set new config

--- a/apstra/client.go
+++ b/apstra/client.go
@@ -1674,6 +1674,9 @@ func (o *Client) GetSystemAgentManagerConfig(ctx context.Context) (*SystemAgentM
 // SetSystemAgentManagerConfig uses a *SystemAgentManagerConfig object to configure the Advanced Settings
 // found on the Managed Devices page of the Web UI.
 func (o *Client) SetSystemAgentManagerConfig(ctx context.Context, cfg *SystemAgentManagerConfig) error {
+	if compatibility.HasDeviceOsImageDownloadTimeout.Check(o.apiVersion) != (cfg.DeviceOsImageDownloadTimeout != nil) {
+		return fmt.Errorf("DeviceOsImageDownloadTimeout is required with apstra %s, and must not be used with other versions", compatibility.HasDeviceOsImageDownloadTimeout)
+	}
 	if !compatibility.SystemManagerHasSkipInterfaceShutdownOnUpgrade.Check(o.apiVersion) && cfg.SkipInterfaceShutdownOnUpgrade {
 		return fmt.Errorf("SkipInterfaceShutdownOnUpgrade may only be used with apstra %s", compatibility.SystemManagerHasSkipInterfaceShutdownOnUpgrade)
 	}

--- a/apstra/compatibility/compatibility.go
+++ b/apstra/compatibility/compatibility.go
@@ -15,6 +15,7 @@ const (
 	apstra422  = "4.2.2"
 	apstra500  = "5.0.0"
 	apstra501  = "5.0.1"
+	apstra510  = "5.1.0"
 )
 
 var (

--- a/apstra/compatibility/constraints.go
+++ b/apstra/compatibility/constraints.go
@@ -17,11 +17,14 @@ var (
 	BpHasVirtualNetworkPolicyNode = Constraint{
 		constraints: version.MustConstraints(version.NewConstraint("<=" + apstra420)),
 	}
+	EmptyVnBindingsOk = Constraint{
+		constraints: version.MustConstraints(version.NewConstraint(">=" + apstra500)),
+	}
 	FabricSettingsApiOk = Constraint{
 		constraints: version.MustConstraints(version.NewConstraint(">=" + apstra421)),
 	}
-	EmptyVnBindingsOk = Constraint{
-		constraints: version.MustConstraints(version.NewConstraint(">=" + apstra500)),
+	HasDeviceOsImageDownloadTimeout = Constraint{
+		constraints: version.MustConstraints(version.NewConstraint(">=" + apstra510)),
 	}
 	IbaDashboardSupported = Constraint{
 		constraints: version.MustConstraints(version.NewConstraint(">=" + apstra500)),

